conan/tools/meson/toolchain.py

```diff
@@ -171,13 +179,12 @@ def _context(self):
             "b_ndebug": self._to_meson_value(self._ndebug) if self._build_type else None,
             # https://mesonbuild.com/Builtin-options.html#compiler-options
             "cpp_std": self._to_meson_cppstd(self._cppstd) if self._cppstd else None,
-            "c_args": self._none_if_empty(self._env.get("CPPFLAGS", '') +
-                                          self._env.get("CFLAGS", '')),
-            "c_link_args": self._env.get("LDFLAGS", None),
-            "cpp_args": self._none_if_empty(self._env.get("CPPFLAGS", '') +
-                                            self._env.get("CXXFLAGS", '')),
-            "cpp_link_args": self._env.get("LDFLAGS", None),
-            "pkg_config_path": "'%s'" % os.getcwd()
+            "c_args": self._to_meson_value(self._env_array('CPPFLAGS') + self._env_array('CFLAGS')),
```

Comment:
New toolchains will provide a different way to manage the environment, please don't use it yet.